plaso/engine/artifact_filters.py

```diff
@@ -0,0 +1,259 @@
+        find_specs[artifact_types.TYPE_INDICATOR_FILE].append(
+            find_spec)
+
+  def BuildFindSpecsFromRegistryArtifact(
+      self, key_entry, find_specs):
+    """Build find specifications from a Windows registry artifact type.
+
+    Args:
+      key_entry (str):  Current file system key to add.
+      find_specs dict[artifacts.artifact_types]:  Dictionary containing
+          find_specs.
+    """
+    for key in self._ExpandRecursiveGlobs(key_entry):
+      if '%%' in key:
+        logging.error((
+            'Unable to expand path filter: {0:s}').format(key))
+        continue
+      find_spec = registry_searcher.FindSpec(key_path_glob=key)
+      if artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY not in find_specs:
+        find_specs[artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY] = []
+
+      find_specs[artifact_types.TYPE_INDICATOR_WINDOWS_REGISTRY_KEY].append(
+          find_spec)
+
+  def _CheckKeyCompatibility(self, key):
+    """Check if a registry key is compatible with dfwinreg.
+
+    Args:
+      key (str):  String key to to check for dfwinreg compatibility.
+
+    Returns:
+      (bool): Boolean whether key is compatible or not.
+    """
+    key_path_prefix = key.split('\\')[0]
+    if key_path_prefix.upper() in self.COMPATIBLE_DFWINREG_KEYS:
+      return True
+    logging.warning('Key {0:s}, has a prefix {1:s} that is not supported '
+                    'by dfwinreg presently'.format(key, key_path_prefix))
+    return False
+
+  def _ExpandRecursiveGlobs(self, path):
+    """Expand recursive like globs present in an artifact path.
+
+    If a path ends in '**', with up to two optional digits such as '**10',
+    the '**' will recursively match all files and zero or more directories
+    from the specified path. The optional digits indicate the recursion depth.
+    By default recursion depth is 10 directories.
+    If the glob is followed by a ‘/’ or '\', only directories and
+    subdirectories will be matched.
+
+    Args:
+      path (str):  String path to be expanded.
+
+    Returns:
+      list[str]: String path expanded for each glob.
+    """
+
+    match = re.search(r'(.*)?(\\|/)\*\*(\d{1,2})?(\\|/)?$', path)
+    if match:
+      skip_first = False
+      if match.group(4):
+        skip_first = True
+      if match.group(3):
+        iterations = match.group(3)
+      else:
+        iterations = self.RECURSIVE_GLOB_LIMIT
+        logging.warning('Path {0:s} contains fully recursive glob, limiting '
+                        'to ten levels'.format(path))
+      paths = self._BuildRecursivePaths(match.group(1), iterations, skip_first)
+      return paths
+    else:
+      return [path]
+
+  def _BuildRecursivePaths(self, path, count, skip_first):
+    """Append wildcard entries to end of path.
+
+    Args:
+      path (str):  Path to append wildcards to.
+
+    Returns:
+      paths list[str]: Paths expanded with wildcards.
+    """
+    paths = []
+    if '\\' in path:
```

Comment:
check the path separator do not blindly assume `\` or `/` are path segment separators also see: https://en.wikipedia.org/wiki/Path_(computing)

Reply:
Done.